### ISSUE TITLE
Firebase deploy functions for Node 8 fails when package-lock.json is present

### ISSUE THREAD

**Issue by motin** [firebase/firebase-tools#1060]
TL:DR - As a follow up to https://github.com/firebase/firebase-tools/issues/947. While it is possible to get the functions deployed by removing package-lock.json, I'd rather not have to omit package-lock.json from that directory since it ideally should ensure that the same versions of all dependencies that was used during local testing also are deployed.

### Version info

Firebase 6.1.2
Node 8.11.1

### Platform Information

OS X

### Steps to reproduce

See https://github.com/firebase/firebase-tools/issues/947 

### Expected behavior

Deploy succeeds even without removing package-lock.json from the functions directory.

### Actual behavior

Deploy fails unless package-lock.json is removed from the functions directory.

**Comment by bkendall** [comment]
In the previous issue, it seems it was never resolved. It looks to me like the `package-lock.json` might just have been out of date after the upgrade from node 6 to 8, which could have caused issues.

I've confirmed that the existence of `package-lock.json` is not blocking my test deploys on node 8. If you have more information and reproduction information, please open another issue with the relevant details. Thanks.

**Comment by motin** [comment]
@bkendall Nuking package-lock.json was the only way I got it to work, merely updating it did not help. Nevertheless, if the previous issue was never resolved, should it be opened, or should a new issue be reported instead?

**Comment by bkendall** [comment]
As previously stated, you will need to provide more detailed information and reproduction instructions so we can take a closer look. Without more information, there's not much help we can provide. Please open a new issue with the relevant details.
